src/libraries/System.Web.HttpUtility/src/System/Web/Util/HttpEncoder.cs

```diff
@@ -212,8 +212,14 @@ internal static string JavaScriptStringEncode(string? value)
                 return null;
             }
 
+            return UrlDecode(bytes.AsSpan(), offset, count);
+        }
+
+
+        internal static byte[] UrlDecode(ReadOnlySpan<byte> bytes, int offset, int count)
```

Comment:
Having a signature that takes both a span and the offset/count is odd.
This should only take a span and the callers are responsible for slicing it correctly.